dashboard/test/models/user_test.rb

```diff
@@ -1685,6 +1685,30 @@ def track_progress(user_id, script_level, result, pairings: nil)
     create :teacher
   end
 
+  test 'admin_changed? equates nil and false' do
+    user = create :user
+    admin = create :admin
+
+    # no change to admin, false
+    refute user.admin_changed?
+    refute admin.admin_changed?
+
+    user.admin = false
+    refute user.admin_changed?
+
+    user.admin = nil
+    refute user.admin_changed?
+
+    user.admin = true
+    assert user.admin_changed?
+
+    admin.admin = nil
+    assert admin.admin_changed?
+
+    admin.admin = false
+    assert admin.admin_changed?
```

Comment:
That `admin_changed?` checks state in comparison to the DB, not history. So mutating admin without saves leads to funniness. For example:
```
irb(main):002:0> u.admin
=> nil
irb(main):003:0> u.admin = true
=> true
irb(main):004:0> u.admin_changed?
=> true
irb(main):005:0> u.admin = false
=> false
irb(main):006:0> u.admin_changed?
=> false
```